rmw_fastrtps_shared_cpp/test/test_names.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include "rmw/qos_profiles.h"
+
+#include "rmw_fastrtps_shared_cpp/names.hpp"
+#include "rmw_fastrtps_shared_cpp/namespace_prefix.hpp"
+
+TEST(NamespaceTest, get_prefix) {
+  EXPECT_EQ("", _get_ros_prefix_if_exists(""));
+  EXPECT_EQ("", _get_ros_prefix_if_exists("not/a_ros_prefix"));
+  for (const auto & prefix : _get_all_ros_prefixes()) {
+    EXPECT_EQ("", _get_ros_prefix_if_exists(prefix + "_should_not_match"));
+    EXPECT_EQ("", _get_ros_prefix_if_exists("this/should_not_match/" + prefix));
```

Comment:
I think forward slash should be at prefix.length(), since that's what all the if statements are checking for. 

```suggestion
    EXPECT_EQ("", _get_ros_prefix_if_exists("th/is_should_not_match/" + prefix));
```